lib/runtime/src/task_spec/op_task_invocation.h

```diff
@@ -48,6 +40,11 @@ struct OpTaskBinding {
   void bind(slot_id, OpTensorSpec const &);
   void bind_grad(slot_id, OpTensorSpec const &);
 
+  template <typename T>
+  void bind(slot_id name, VariadicTensorRef<T> const &t) {
```

Comment:
Why not just `bind(slot_id start, std::vector<OpTensorSpec> const &)`? And then the value type of the `tensor_bindings` map can just be `variant<OpTensorSpec, std::vector<OpTensorSpec>>`? 

Either that, or we can maintain a separate mapping for variadic.